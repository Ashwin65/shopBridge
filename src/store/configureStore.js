import { applyMiddleware, combineReducers, compose, createStore } from 'redux';
import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import createSagaMiddleware from 'redux-saga';
import rootSaga from '../redux/sagas/sagas';
import {listItemReducer} from '../redux/reducers/listItemReducer'


const sagaMiddleware = createSagaMiddleware();
const rootReducer = combineReducers({
      Items: listItemReducer,
});
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const persistConfig = {
  key: 'root',
  storage,
};
const persistedReducer = persistReducer(persistConfig, rootReducer);


let store = createStore(
  rootReducer,
  composeEnhancers(applyMiddleware(sagaMiddleware))
);

sagaMiddleware.run(rootSaga);

export { store };
