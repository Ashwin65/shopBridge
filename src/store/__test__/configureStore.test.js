import { store } from '../configureStore';

describe('Testing configure store', () => {
  it('Should match the snapshot', () => {
    const container = store;
    expect(container).toMatchSnapshot();
  });
});
  