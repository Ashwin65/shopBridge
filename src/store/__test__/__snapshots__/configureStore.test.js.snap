// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing configure store Should match the snapshot 1`] = `
Object {
  "@@observable": [Function],
  "dispatch": [Function],
  "getState": [Function],
  "replaceReducer": [Function],
  "subscribe": [Function],
}
`;
